### DISCUSSION TITLE
comint mode / general repl support

### DISCUSSION THREAD

**Discussion by bo-tato** [lem-project/lem#1103] [General]
As for as I can tell with a little searching lem currently just has a repl for lisp? Is there anything comparable to comint mode that would enable quickly developing repl support for python, ruby, sql, etc?

**Comment by bo-tato** [comment]
I should search the source rather the just searching github :smile:
it seems the equivalent of comint would be lem/listener-mode which is used for a python and elixir repls

**Comment by bo-tato** [comment]
Source seems to be straightforward and readable! It is a little concerning though there are almost no docstrings. A great strength of emacs is how thoroughly documented everything is. Manual and guide style documentation doesn't have to be responsibility of programmer, others can do it later, but imo docstrings should be written by the coder when they write the function.

**Reply by seanfarley** [reply]
Pull requests are welcomed! I agree that documentation is great but lem is still in its early stages and there are hurdles such as a quickly evolving refactoring and things such as language barriers. Once things stabilize a bit more, I plan to do a pass of adding some docstrings and documentation.

**Reply by bo-tato** [reply]
thanks! It is really tempting to get into lem development but for now I don't have time for it and stick with emacs
